internal/parser/scanner/states.go

```diff
@@ -42,197 +182,137 @@ var keywordsWithC map[string]token.Type = map[string]token.Type{
+	"JOIN":              token.KeywordJoin,
+	"KEY":               token.KeywordKey,
+	"LAST":              token.KeywordLast,
+	"LEFT":              token.KeywordLeft,
+	"LIKE":              token.KeywordLike,
+	"LIMIT":             token.KeywordLimit,
+	"MATCH":             token.KeywordMatch,
+	"NATURAL":           token.KeywordNatural,
+	"NO":                token.KeywordNo,
+	"NOT":               token.KeywordNot,
+	"NOTHING":           token.KeywordNothing,
+	"OF":                token.KeywordOf,
+	"OFFSET":            token.KeywordOffset,
+	"ON":                token.KeywordOn,
+	"OR":                token.KeywordOr,
+	"ORDER":             token.KeywordOrder,
+	"OTHERS":            token.KeywordOthers,
+	"OUTER":             token.KeywordOuter,
+	"OVER":              token.KeywordOver,
+	"PARTITION":         token.KeywordPartition,
+	"PLAN":              token.KeywordPlan,
+	"PRAGMA":            token.KeywordPragma,
+	"PRECEDING":         token.KeywordPreceding,
+	"PRIMARY":           token.KeywordPrimary,
+	"QUERY":             token.KeywordQuery,
+	"RAISE":             token.KeywordRaise,
+	"RANGE":             token.KeywordRange,
+	"RECURSIVE":         token.KeywordRecursive,
+	"REFERENCES":        token.KeywordReferences,
+	"REGEXP":            token.KeywordRegexp,
+	"REINDEX":           token.KeywordReindex,
+	"RELEASE":           token.KeywordRelease,
+	"RENAME":            token.KeywordRename,
+	"REPLACE":           token.KeywordReplace,
+	"RESTRICT":          token.KeywordRestrict,
+	"RIGHT":             token.KeywordRight,
+	"ROLLBACK":          token.KeywordRollback,
+	"ROW":               token.KeywordRow,
+	"ROWS":              token.KeywordRows,
+	"SAVEPOINT":         token.KeywordSavepoint,
+	"SELECT":            token.KeywordSelect,
+	"SET":               token.KeywordSet,
+	"TABLE":             token.KeywordTable,
+	"TEMP":              token.KeywordTemp,
+	"TEMPORARY":         token.KeywordTemporary,
+	"THEN":              token.KeywordThen,
+	"TIES":              token.KeywordTies,
+	"TO":                token.KeywordTo,
+	"TRANSACTION":       token.KeywordTransaction,
+	"TRIGGER":           token.KeywordTrigger,
+	"UNBOUNDED":         token.KeywordUnbounded,
+	"UNION":             token.KeywordUnion,
+	"UNIQUE":            token.KeywordUnique,
+	"UPDATE":            token.KeywordUpdate,
+	"USING":             token.KeywordUsing,
+	"VACUUM":            token.KeywordVacuum,
+	"VALUES":            token.KeywordValues,
+	"VIEW":              token.KeywordView,
+	"VIRTUAL":           token.KeywordVirtual,
+	"WHEN":              token.KeywordWhen,
+	"WHERE":             token.KeywordWhere,
+	"WINDOW":            token.KeywordWindow,
+	"WITH":              token.KeywordWith,
+	"WITHOUT":           token.KeywordWithout,
+}
+
+var (
+	whiteSpace            = matcher.Merge(formFeed, noBreakSpace, space, horizontalTab, unicodeSpace, verticalTab, zeroWidthJoiner, zeroWidthNoBreakSpace, zeroWidthNonJoiner)
```

Comment:
This is the whitespace definition of `ECMAScript`. I sent this to you as an example for the usage of `matcher`s. This is not the `SQL` definition of a whitespace.

Reply:
Sure, Ill look into it.